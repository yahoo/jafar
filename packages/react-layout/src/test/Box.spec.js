/**
  * Copyright 2020, Verizon Media
  * Licensed under the terms of the MIT license. See LICENSE file in project root for terms.
  */

import React from 'react';
import { shallow } from 'enzyme';
import Box from '../components/Box/index';

describe('Box', () => {
  const Field = () => {};

  it('Should render component - ok', () => {
    const box = {
      component: () => {},
      props: { id: 'firstName' },
    };
    const component = shallow(getComponent(box));
    expect(component).toMatchSnapshot();
  });
  it('Should render row - ok', () => {
    const box = {
      direction: 'row',
      boxes: [{
        component: Field,
        props: { id: 'firstName' },
      }, {
        component: Field,
        props: { id: 'lastName' },
      }],
    };
    const component = shallow(getComponent(box));
    expect(component).toMatchSnapshot();
  });

  it('Should render column, ok', () => {
    const box = {
      direction: 'column',
      boxes: [{
        component: Field,
        props: { id: 'firstName' },
      }, {
        component: Field,
        props: { id: 'lastName' },
      }],
    };
    const component = shallow(getComponent(box));
    expect(component).toMatchSnapshot();
  });

  it('Should render nested content, ok', () => {
    const content = {
      direction: 'row',
      content: [{
        direction: 'column',
        boxes: [{
          component: Field,
          props: { id: 'firstName' },
        }, {
          component: Field,
          props: { id: 'lastName' },
        }],
      }, {
        direction: 'column',
        boxes: [{
          component: Field,
          props: { id: 'address' },
        }],
      }],
    };
    const component = shallow(getComponent(content));
    expect(component).toMatchSnapshot();
  });

  it('Should render with style, ok', () => {
    const box = {
      direction: 'column',
      style: {
        width: '300px',
        maxWidth: '500px',
        margin: '10px 20px',
      },
      boxes: [{
        component: Field,
        props: { id: 'firstName' },
      }, {
        component: Field,
        props: { id: 'lastName' },
      }],
    };
    const component = shallow(getComponent(box));
    expect(component).toMatchSnapshot();
  });

  function getComponent(props) {
    return (<Box {...props} />);
  }
});
