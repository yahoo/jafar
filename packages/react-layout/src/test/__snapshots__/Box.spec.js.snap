// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box Should render column, ok 1`] = `
<styled.div>
  <Box
    boxes={Array []}
    component={[Function]}
    direction="column"
    key="0"
    props={
      Object {
        "id": "firstName",
      }
    }
    style={Object {}}
  />
  <Box
    boxes={Array []}
    component={[Function]}
    direction="column"
    key="1"
    props={
      Object {
        "id": "lastName",
      }
    }
    style={Object {}}
  />
</styled.div>
`;

exports[`Box Should render component - ok 1`] = `
<component
  id="firstName"
/>
`;

exports[`Box Should render nested content, ok 1`] = `<styled.div />`;

exports[`Box Should render row - ok 1`] = `
<styled.div>
  <Box
    boxes={Array []}
    component={[Function]}
    direction="column"
    key="0"
    props={
      Object {
        "id": "firstName",
      }
    }
    style={Object {}}
  />
  <Box
    boxes={Array []}
    component={[Function]}
    direction="column"
    key="1"
    props={
      Object {
        "id": "lastName",
      }
    }
    style={Object {}}
  />
</styled.div>
`;

exports[`Box Should render with style, ok 1`] = `
<styled.div
  margin="10px 20px"
  maxWidth="500px"
  width="300px"
>
  <Box
    boxes={Array []}
    component={[Function]}
    direction="column"
    key="0"
    props={
      Object {
        "id": "firstName",
      }
    }
    style={Object {}}
  />
  <Box
    boxes={Array []}
    component={[Function]}
    direction="column"
    key="1"
    props={
      Object {
        "id": "lastName",
      }
    }
    style={Object {}}
  />
</styled.div>
`;
